crates/cxx-qt-gen/src/generator/cpp/property/setter.rs

```diff
@@ -6,32 +6,39 @@
 use crate::generator::{cpp::fragment::CppFragment, naming::property::QPropertyNames};
 use indoc::formatdoc;
 
-pub fn generate(idents: &QPropertyNames, qobject_ident: &str, cxx_ty: &str) -> CppFragment {
-    CppFragment::Pair {
-        header: format!(
-            "Q_SLOT void {ident_setter}({cxx_ty} const& value);",
-            ident_setter = idents.setter.cxx_unqualified(),
-        ),
-        source: formatdoc! {
-            r#"
-            void
-            {qobject_ident}::{ident_setter}({cxx_ty} const& value)
-            {{
-                const ::rust::cxxqt1::MaybeLockGuard<{qobject_ident}> guard(*this);
-                {ident_setter_wrapper}(value);
-            }}
-            "#,
-            ident_setter = idents.setter.cxx_unqualified(),
-            ident_setter_wrapper = idents.setter_wrapper.cxx_unqualified(),
-        },
+pub fn generate(idents: &QPropertyNames, qobject_ident: &str, cxx_ty: &str) -> Option<CppFragment> {
+    // Checking whether a setter should be generated based off if it has a name
+    if let (Some(setter), Some(setter_wrapper)) = (&idents.setter, &idents.setter_wrapper) {
```

Comment:
as we were discussing maybe the enum route is better?

This would then become
`if let (QPropertyFlagState::Auto(setter), QPropertyFlagState::Auto(setter_wrapper) = (&idents.setter, &idents.setter_wrapper) {`

Which would then mean that this generate function could be called even when a custom setter is set, as that is a different enum, which then removes that if condition on the flag in the other file

Reply:
Yeah exactly, or still using Some() if going for Option<State> as mentioned above

Reply:
done